web3/contract.py

```diff
@@ -808,14 +826,23 @@ def __getattr__(self, attr: Any) -> Callable[[], None]:
         return NonExistentFallbackFunction._raise_exception
 
 
+class NonExistentReceiveFunction:
+    @staticmethod
+    def _raise_exception() -> NoReturn:
+        raise FallbackNotFound("No receive function was found in the contract ABI.")
+
+    def __getattr__(self, attr: Any) -> Callable[[], None]:
+        return NonExistentReceiveFunction._raise_exception
```

Comment:
Does `return self._raise_exception` not work?

Reply:
It does. I was just following the pattern from `NonExistentFallbackFunction` above. Will change 👍